### PR TITLE
Add seamless-immutable type definition file

### PR DESCRIPTION
**Pull request by alex3165**
case 1. Add a new type definition.
- [ ] checked compilation succeeds with `--target es6` and `--noImplicitAny` options.
- [ ] has correct [naming convention](http://definitelytyped.org/guides/contributing.html#naming-the-file)
- [ ] has a [test file](http://definitelytyped.org/guides/contributing.html#tests) with the suffix of  `-tests.ts` or `-tests.tsx`.

case 2. Improvement to existing type definition.
- documentation or source code reference which provides context for the suggested changes.  url http://api.jquery.com/html .
  - it has been reviewed by a DefinitelyTyped member.

### CODE REVIEW

File: seamless-immutable/seamless.d.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
+// Type definitions for Seamless-immutable 6.1.3
+// Project: https://github.com/rtfeldman/seamless-immutable
```
Comment (mhegazy) [inline seamless-immutable/seamless.d.ts]: the file name should be `seamless-immutable\seamless-immutable.d.ts`

File: seamless-immutable/seamless.d.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+// Definitions by: alex3165 <https://github.com/alex3165>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+// Copyright (c) 2016, Richard Feldman
```
Comment (mhegazy) [inline seamless-immutable/seamless.d.ts]: For legal reasons, we can not have different licence text for packages. all packages use the MIT licence under: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/LICENSE

Comment (mhegazy) [inline seamless-immutable/seamless.d.ts]: So we will not be able to accept this PR as long as it has this specialized licence text.

File: seamless-immutable/seamless.d.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    set(property: string, value: any): ImmutableObject<any>;
+    setIn(propertyPath: Array<string>, value: any): Immutable<any>;
+
+    asMutable(): T;
+    asMutable(opts: AsMutableOptions): T;
+
+    merge(part: any, config?: MergeConfig): ImmutableObject<T>;
+
+    update(property: string, updaterFunction: (value: any) => any): ImmutableObject<T>;
+    update(property: string, updaterFunction: (value: any, additionalParameter1: any) => any, arg1: any): ImmutableObject<T>;
```
Comment (mhegazy) [inline seamless-immutable/seamless.d.ts]: I would recommend changing these into one signature with rest args:

```ts
update(property: string, updaterFunction: (value: any, ...additionalParamters: any[]) => any, ...additionalArguments: any[]): ImmutableObject<T>;
```

File: seamless-immutable/seamless.d.ts [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    update(property: string, updaterFunction: (value: any, additionalParameter1: any, additionalParameter2: any, additionalParameter3: any, additionalParameter4: any) => any, arg1: any, arg2: any, arg3: any, arg4: any): ImmutableObject<T>;
+
+    updateIn(propertyPath: Array<string>, updaterFunction: (value: any) => any): ImmutableObject<T>;
+    updateIn(propertyPath: Array<string>, updaterFunction: (value: any, additionalParameter1: any) => any, arg1: any): ImmutableObject<T>;
+    updateIn(propertyPath: Array<string>, updaterFunction: (value: any, additionalParameter1: any, additionalParameter2: any) => any, arg1: any, arg2: any): ImmutableObject<T>;
+    updateIn(propertyPath: Array<string>, updaterFunction: (value: any, additionalParameter1: any, additionalParameter2: any, additionalParameter3: any) => any, arg1: any, arg2: any, arg3: any): ImmutableObject<T>;
+    updateIn(propertyPath: Array<string>, updaterFunction: (value: any, additionalParameter1: any, additionalParameter2: any, additionalParameter3: any, additionalParameter4: any) => any, arg1: any, arg2: any, arg3: any): ImmutableObject<T>;
+
+    without(property: string): ImmutableObject<any>;
```
Comment (mhegazy) [inline seamless-immutable/seamless.d.ts]: this is already covered by `without(...properties: string[]): ImmutableObject<any>;`

**Review by mhegazy — CHANGES_REQUESTED** [review]

**Comment by alex3165** [comment]
@mhegazy Thank you for your advice I have made the changes.

**Review by mhegazy — APPROVED** [review]

**Comment by mhegazy** [comment]
thanks. can you please address the failing CI tests.

**Comment by alex3165** [comment]
@mhegazy Fixed

**Comment by toranb** [comment]
Anyone have an example app using this w/ TS 2x ? Just looking around for some samples to see how you would import and use use something like `SI.ImmutableObject` with ES2015+ syntax and TS 2

**Pull request merged**